### PR TITLE
feat(swap): Store deposit transaction ref on swap [WEB-1189]

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: packages/swap/src/event-handlers/networkBroadcastSuccess.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const networkBroadcastSuccessArgs = (chain: Chain) =>
           .transform((v) => `${v.blockNumber}-${v.extrinsicIndex}`),
         hexString,
       ])
-      .transform((v) => (chain === 'Bitcoin' ? reverseBytes(v) : v))
+      .transform((v) => (chain === 'Bitcoin' ? reverseBytes(v.substring(2)) : v))
```
Comment (niklasnatter) [inline packages/swap/src/event-handlers/networkBroadcastSuccess.ts:17]: we exposed the bitcoin tx hash with a leading `0x` till now (bitcoin explorers do not support that when i tested) 😕 will add a migration for the existing broadcasts

File: packages/sdk/src/swap/types.ts [hunk 2]
```diff
@@ -129,7 +129,8 @@ type SwapState =
   | {
       state: 'AWAITING_DEPOSIT';
       depositAmount: string | undefined;
-      depositTransactionHash: string | undefined;
+      depositTransactionHash: string | undefined; // DEPRECATED(1.5): use depositTransactionRef instead
```
Comment (niklasnatter) [inline packages/sdk/src/swap/types.ts:132]: unfortuante existing naming bc the polkadot thing is not a hash (and we call it `ref` for broadcasts). will expose both names for now

**Review by acdibble — APPROVED** [review]

File: packages/swap/prisma/migrations/20240711155734_migrate_bitcoin_transaction_ref/migration.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+UPDATE "public"."Broadcast" SET "transactionRef" = SUBSTRING("transactionRef", 3) WHERE "chain" = 'Bitcoin' AND "transactionRef" IS NOT NULL;
```
Comment (jerryafr) [inline packages/swap/prisma/migrations/20240711155734_migrate_bitcoin_transaction_ref/migration.sql:1]: is 3 correct here? shouldn't this be substring( ..., 2) ?

**Review by jerryafr — APPROVED** [review]

File: packages/swap/prisma/migrations/20240711155734_migrate_bitcoin_transaction_ref/migration.sql (hunk 3 above)
Comment (niklasnatter) [reply packages/swap/prisma/migrations/20240711155734_migrate_bitcoin_transaction_ref/migration.sql:1]: surprisingly yes - if you pass 1 the first character will still be included 🙈

File: packages/swap/src/event-handlers/v120/networkDepositReceived.ts [hunk 4]
```diff
@@ -1,19 +1,58 @@
...
+  evmDepositDetails,
+  polkadotDepositDetails,
+]);
+
 export const depositReceivedArgs = z.object({
   amount: u128,
   asset: internalAssetEnum,
   ingressFee: u128,
   action: actionSchema,
+  blockHeight: u128.optional(),
```
[the remark this answers is not in the document]
Comment (niklasnatter) [reply packages/swap/src/event-handlers/v120/networkDepositReceived.ts:25]: yes that makes sense!

**Pull request merged**